contrib/net/http/trace.go

```diff
@@ -63,24 +77,23 @@ func TraceAndServe(h http.Handler, w http.ResponseWriter, r *http.Request, cfg *
 	if spanctx, err := tracer.Extract(tracer.HTTPHeadersCarrier(r.Header)); err == nil {
 		opts = append(opts, tracer.ChildOf(spanctx))
 	}
-	span, ctx := tracer.StartSpanFromContext(r.Context(), "http.request", opts...)
-	rw, ddrw := wrapResponseWriter(w)
-	defer func() {
-		if ddrw.status == 0 {
-			span.SetTag(ext.HTTPCode, "200")
-		} else {
-			span.SetTag(ext.HTTPCode, strconv.Itoa(ddrw.status))
-		}
-		if ddrw.status >= 500 && ddrw.status < 600 {
-			span.SetTag(ext.Error, fmt.Errorf("%d: %s", ddrw.status, http.StatusText(ddrw.status)))
-		}
-		span.Finish(cfg.FinishOpts...)
-	}()
+	return tracer.StartSpanFromContext(r.Context(), "http.request", opts...)
+}
 
-	if appsec.Enabled() {
-		h = httpsec.WrapHandler(h, span, cfg.RouteParams)
+// FinishRequestSpan finishes the given HTTP request span with the standard list of HTTP request span tags.
```

Comment:
Let's be clearer about what "finishes" means here. What exactly is it doing?

Reply:
Added a small clarification in https://github.com/DataDog/dd-trace-go/pull/1286/commits/8d9bd76eb8b97263ace3e18f76774f6a74bb16c3

Note that the documentation of [`tracer.(Span).Finish()`](https://pkg.go.dev/gopkg.in/DataDog/dd-trace-go.v1@v1.38.1/ddtrace#Span)  doesn't say give much more details but that's where I would put extra details (out of the scope of this PR though).

WDYT?

Reply:
I actually think the previous version was a bit better. I was thinking more about an explanation of the status strings and errors in the tags, but no biggie either way.

Reply:
Reverted in https://github.com/DataDog/dd-trace-go/pull/1286/commits/92607eff5eb294c9c25aabb2540543797efd8592